### PR TITLE
Enable edge.forward to be traced

### PR DESCRIPTION
**Pull request by johannes-no**
# Issue
For a better accuracy of our services we need more precise information about the edges that where choosen on the rute.
Especially in which direction the edge was traversed.

# Fix
With this pull request we enable the "trace_attributes" endpoint to retrieve information about in which direction the edge was traversed.

# Relations
This PR is based on [4823](https://github.com/valhalla/valhalla/pull/4825), the changes are the same. 

# Tasklist
- [x] Update the docs with any new request parameters or changes to behavior described
- [x] Update the [changelog](https://github.com/valhalla/valhalla/pull/CHANGELOG.md)

Johannes Nonnenmacher [johannes.nonnenmacher@mercedes-benz.com](mailto:johannes.nonnenmacher@mercedes-benz.com), Mercedes-Benz Tech Innovation GmbH
[Provider Information](https://github.com/mercedes-benz/foss/blob/master/PROVIDER_INFORMATION.md)

Licensed under MIT

### CODE REVIEW

**Review by chrstnbwnkl — CHANGES_REQUESTED** [review]
Just a tiny nit: add a `#` before the PR number in the changelog entry, and a space between the entry description and the url :smile:

**Comment by johannes-no** [comment]
No problem, I was so free and adjusted it for all entries 😄

File: CHANGELOG.md [hunk 1]
```diff
@@ -146,9 +146,10 @@
    * ADDED: `expansion_type` property to `/expansion` [#4784](https://github.com/valhalla/valhalla/pull/4784)
    * ADDED: inline config arg for `valhalla_build_elevation` script [#4787](https://github.com/valhalla/valhalla/pull/4787)
    * ADDED: `use_truck_route` [#4809](https://github.com/valhalla/valhalla/pull/4809)
-   * ADDED: Add option `edge.country_crossing` to trace attributes[4825](https://github.com/valhalla/valhalla/pull/4825)
-   * CHANGED: Unification of turn costs for ramps and roundabouts[4827](https://github.com/valhalla/valhalla/pull/4827)
+   * ADDED: Add option `edge.country_crossing` to trace attributes [#4825](https://github.com/valhalla/valhalla/pull/4825)
+   * CHANGED: Unification of turn costs for ramps and roundabouts [#4827](https://github.com/valhalla/valhalla/pull/4827)
    * CHANGED: updated dockerfile to use ubuntu 24.04 [#4805](https://github.com/valhalla/valhalla/pull/4805)
+   * ADDED: Add option `edge.forward` to trace attributes [#4876](https://github.com/valhalla/valhalla/pull/4876)
```
Comment (johannes-no) [inline CHANGELOG.md:152]: This is the newly added entry in the change log

**Comment by chrstnbwnkl** [comment]
Ah crap, one more thing: the new entry is listed now as part of the 3.5.0 release, should be under the unreleased section. Sorry, just noticed it now...

File: docs/docs/api/map-matching/api-reference.md [hunk 2]
```diff
@@ -220,7 +221,8 @@ Each `edge` may include:
...
+| `country_crossing` | True if the edge is a country crossing. |                                                                                                                                                                                                                                                                                                                                                                                                                             |
+| `forward` | True if the edge is traversed forwards and False if it is traversed backwars.                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (chrstnbwnkl) [inline docs/docs/api/map-matching/api-reference.md:225]: One more thing: backwars should be backwards :smile:

**Review by chrstnbwnkl — CHANGES_REQUESTED** [review]

**Review by chrstnbwnkl — APPROVED** [review]

**Comment by chrstnbwnkl** [comment]
Should be good now, thanks!

File: src/thor/triplegbuilder.cc [hunk 3]
```diff
@@ -1206,6 +1206,11 @@ TripLeg_Edge* AddTripEdge(const AttributesController& controller,
     trip_edge->set_country_crossing(directededge->ctry_crossing());
   }
 
+  // Set forward if requested
+  if (controller(kEdgeForward)) {
+    trip_edge->set_forward(directededge->forward());
```
Comment (kevinkreiser) [inline src/thor/triplegbuilder.cc:1211]: you guys know that this is the direction of the shape not the "traversal" direction

File: docs/docs/api/map-matching/api-reference.md [hunk 4]
```diff
@@ -220,7 +221,8 @@ Each `edge` may include:
...
+| `country_crossing` | True if the edge is a country crossing. |                                                                                                                                                                                                                                                                                                                                                                                                                             |
+| `forward` | True if the edge is traversed forwards and False if it is traversed backwards.                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (kevinkreiser) [inline docs/docs/api/map-matching/api-reference.md:225]: `| `forward` | True if the edge is traversed forwards and False if it is traversed backwards with respect to the reference shape/geometry (ie. the direction in which it was digitized).`

Comment (kevinkreiser) [reply docs/docs/api/map-matching/api-reference.md:225]: ill just put this in master and if you were hoping to know if you traversed the edge against the way "access" is allowed then we need to do something a bit more

**Pull request merged**